packages/components/tag/src/Tag.stories.tsx

```diff
@@ -43,11 +44,17 @@ export const Intent: StoryFn = _args => (
   <div className="flex flex-col gap-md">
     {designs.map(design => (
       <div key={design} className="flex flex-row gap-md">
-        {intents.map(intent => (
-          <Tag key={intent} design={design} intent={intent}>
-            {intent} tag
-          </Tag>
-        ))}
+        {intents.map(intent => {
+          if (design !== 'filled' && intent === 'surface') {
+            return <span className="self-center text-small">N/A</span>
+          }
+
+          return (
+            <Tag key={intent} design={design} intent={intent as any}>
```

Comment:
Sorry about the `any` 😅, but without it, TS will complain because it can't infer that due to my previous `if` block, we cannot have a surface of **intent**, with a design of **outlined** or **tinted** when we end up here

Reply:
You should be ashamed!